lib/cli-engine/formatters/stylish.js

```diff
@@ -22,6 +21,51 @@ function pluralize(word, count) {
     return (count === 1 ? word : `${word}s`);
 }
 
+/**
+ * Formats a two-dimensional array of strings into a table.
+ * @param {string[][]} rows The rows of the table, each row being an array of strings representing the columns.
+ * @returns {string} The formatted table.
+ */
+function table(rows) {
+    const columnSizes = new Array(rows[0].length).fill(0);
+
+    for (const row of rows) {
+        for (let columnIndex = 0; columnIndex < row.length; columnIndex++) {
+            const length = stripAnsi(row[columnIndex]).length;
+
+            if (length > columnSizes[columnIndex]) {
+                columnSizes[columnIndex] = length;
+            }
+        }
+    }
+
+    const hsep = "  ";
+    let result = "";
+
+    for (let rowIndex = 0; rowIndex < rows.length; rowIndex++) {
+        const row = rows[rowIndex];
+
+        for (let columnIndex = 0; columnIndex < row.length; columnIndex++) {
+            const column = row[columnIndex];
+            const spacing = " ".repeat(columnSizes[columnIndex] - stripAnsi(column).length);
+
+            result += hsep;
+
+            if (columnIndex === 0) {
+                result += spacing + column;
+            } else {
+                result += column + (columnIndex === row.length - 1 ? "" : spacing);
+            }
+        }
+
+        if (rowIndex < rows.length - 1) {
+            result += "\n";
+        }
```

Comment:
This looks very similar to what an Array `.join("\n")` would look like. You could achieve the same thing with roughly...


```js
return rows
  .map((row, rowIndex) => {
    // ...
  })
  .join("\n")
```

I think doing this would make the code more clear - such as making it more apparent that the `.join("\n")` below is unnecessary. WDYT?

Reply:
I'd like to avoid using `map()` altogether as it introduces a new array under the hood. Handling the newlines by appending them directly increases performance, even if it's not as readable.

That said, I am considering always unconditionally appending a newline at the end stripping the last one after the loop, that removes a condition in the `for` loop, which might boost performance even more.